### PR TITLE
GUI Tool

### PR DESCRIPTION
**Pull request by dakom**
This was previously broken down into different PRs, but it got a bit confusing since they aren't being reviewed/merged in order

Closing out that mess and just having this one giant PR. Happy to jump on to co-review with anyone, should be _fairly_ quick if we ignore the gui itself since the changes in https://github.com/Lay3rLabs/avs-toolkit/pull/80 were already reviewed, and the overall idea follows the pattern of moving code that's used by both gui and cli into shared

A few fixes and changes needed to be done to satisfy the compiler warnings etc.

TO TEST

* as of right now, only way is to test locally
* the layer-sdk repo doesn't have the newest wasmatic. Need to stop the inner wasmatic service in docker and manually run latest wasmatic, with `cargo run up --config=wasmatic-local.toml`

### CODE REVIEW

File: contracts/oracle-verifier/src/msg.rs [hunk 1]
```diff
@@ -42,17 +37,17 @@ pub enum ExecuteMsg {
 #[cw_orch(disable_fields_sorting)]
 #[derive(QueryResponses)]
 pub enum QueryMsg {
-    #[returns(Config)]
+    #[returns(crate::state::Config)]
     Config {},
     /// Ordered by completion time descending (last completed first)
-    #[returns(Option<TaskInfoResponse>)]
+    #[returns(Option<lavs_apis::verifier_simple::TaskInfoResponse>)]
```
Comment (ethanfrey) [inline contracts/oracle-verifier/src/msg.rs:43]: Can you explain why these changes are needed?

File: tools/shared/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[package]
+name = "avs-toolkit-shared"
```
Comment (ethanfrey) [inline tools/shared/Cargo.toml:2]: Okay, this is all logic shared between cli and gui?

**Review by ethanfrey — APPROVED** [review]
I don't really care about correctness of the gui now, as it is experimental tool. And I have no time to review it.

I looked at the other code, and main concerns are the changes to the contract code (why?) as well as if this breaks anything in CLI.

If we could minimize any changes to contracts, rebase on latest main, and manually test the CLI, then I will be happy to merge this and tag

File: contracts/oracle-verifier/src/msg.rs (hunk 1 above)
Comment (dakom) [reply contracts/oracle-verifier/src/msg.rs:43]: funny timing, I think I was writing on Slack literally as you were writing here :)

it's not really gui-specific, it's a warning generated whenever using the contract as a library, for its types, because then it's only used as a schema and not referenced anywhere else so it generates "unused variable" warnings

**Comment by dakom** [comment]
@ethanfrey I got it green again and confirmed that following the CLI instructions still works to:

1. build contracts with optimizer.sh
2. deploy contracts with CLI
3. add a task with CLI
4. view task queue with CLI

**Comment by dakom** [comment]
@ethanfrey based on what you said [above](https://github.com/Lay3rLabs/avs-toolkit/pull/90#pullrequestreview-2384497741) I think that's all fulfilled - so I'll go ahead and merge so you can tag at your convenience.

(there are some other gui changes that will happen later and I'll need to check/fix the pages build - but that's all lower priority than faucet, so it shouldn't be tacked onto blocking this imho, i.e. might not get to those today at all)
